apps/rand.c

```diff
@@ -53,7 +53,7 @@ int rand_main(int argc, char **argv)
     char *outfile = NULL, *prog;
     OPTION_CHOICE o;
     int format = FORMAT_BINARY, r, i, ret = 1, buflen = 131072;
```

Comment:
While you're at it, would you mind replacing the magic number '131072' by something more expressive? 
```suggestion
    int format = FORMAT_BINARY, r, i, ret = 1;
    size_t buflen = 0x20000; /* = 128 kiB */
```
(And also change the type to `size_t`?)

Reply:
The DRBGs are limited to 1 << 16 bytes per request.  Ideally, wed query the RNG about the max request it can satisfy and use that.

Reply:
Um, 1 << 16 is 65535, so isn't buflen already too big by a factor of 2 the way its currently written?

Reply:
IIRC, RAND_bytes does not have this limitation, because somewhere down the call stack the DRBG is called in a loop to generate chunks which don't exceed the maximum allowed size for generate requests.

Reply:
But @paulidale is probably right, it's not ideal if the `openssl rand` app uses a chunk size which is larger than the internal chunk size of `RAND_bytes`, because then chunking happens twice.

Reply:
Ok, PR updated.  We now allow for an arbitrary number of bytes (up to (2^32)-1), or for much larger values by scaling that value with a k|m|g suffix

Reply:
I've corrected rand_main to only request 1<<16 bytes per chunk.  It would still be nice to query the drbg for its max size, but doing so seems like it would introduce the need for additional code to fetch an EVP_RAND ctx, and that seems like a large enough change, that it would warrant a separate pr